crates/lang/macro/src/lib.rs

```diff
@@ -18,20 +18,600 @@ mod trait_def;
+///             value: bool,
+///         }
+///         # impl Flipper {
+///         #     #[ink(constructor)]
+///         #     pub fn construct() -> Self { Flipper { value: false } }
+///         #     #[ink(message)]
+///         #     pub fn message(&self) {}
+///         # }
+///     }
+///     ```
+///
+/// - There must be at least one `#[ink(constructor)]` defined method.
+///
+///     Methods flagged with `#[ink(constructor)]` are special in that they are dispatchable
+///     upon contract instantiation. A contract may define multiple such constructors which
+///     allow users of the contract to instantiate a contract in multiple different ways.
+///
+///     **Example:**
+///
+///     Given the `Flipper` contract definition above we add an `#[ink(constructor)]`
+///     as follows:
+///
+///     ```
+///     # use ink_lang as ink;
+///     # #[ink::contract]
+///     # mod flipper {
+///         # #[ink(storage)]
+///         # pub struct Flipper {
+///         #     value: bool,
+///         # }
+///     impl Flipper {
+///         #[ink(constructor)]
+///         pub fn new(initial_value: bool) -> Self {
+///             Flipper { value: false }
+///         }
+///         # #[ink(message)]
+///         # pub fn message(&self) {}
+///     }
+///     # }
+///     ```
+///
+/// - There must be at least one `#[ink(message)]` defined method.
+///
+///     Methods flagged with `#[ink(message)]` are special in that they are dispatchable
+///     upon contract invokation. The set of ink! messages defined for an ink! smart contract
+///     define the smart contract's API surface with which users are allowed to interact.
+///
+///     An ink! smart contract can have multiple such ink! messages defined.
+///
+///     **Note:**
+///
+///     - An ink! message with a `&self` receiver may only read state whereas an ink! message
+///       with a `&mut self` receiver may mutate the contract's storage.
+///
+///     **Example:**
+///
+///     Given the `Flipper` contract definition above we add some `#[ink(message)]` definitions
+///     as follows:
+///
+///     ```
+///     # use ink_lang as ink;
+///     # #[ink::contract]
+///     # mod flipper {
+///         # #[ink(storage)]
+///         # pub struct Flipper {
+///         #     value: bool,
+///         # }
+///     impl Flipper {
+///         # #[ink(constructor)]
+///         # pub fn new(initial_value: bool) -> Self {
+///         #     Flipper { value: false }
+///         # }
+///         /// Flips the current value.
+///         #[ink(message)]
+///         pub fn flip(&mut self) {
+///             self.value = !self.value;
+///         }
+///
+///         /// Returns the current value.
+///         #[ink(message)]
+///         pub fn get(&self) -> bool {
+///             self.value
+///         }
+///     }
+///     # }
+///     ```
+///
+///     **Payable Messages:**
+///
+///     An ink! message by default will reject calls that additional fund the smart contract.
```

Comment:
I don't get this sentence, do you mean "that additionally fund the smart contract"?

Reply:
yes, good catch!

Reply:
I don't yet fully get it, could you elaborate on what it means that a call additionally funds the smart contract?

Reply:
When you submit a transaction to call a smart contract you can not only set gas limit but also how much actual value you want to transfer to the call.